### PR TITLE
Fixes image_size parameter for training, works for powers of 2 >= 8

### PR DESCRIPTION
**Pull request by md5wasp**
I was not able to get the image_size parameter working when training, it seemed like the code had some hardcoded assumptions about the image size in the generator.

This PR fixes that, but only for cases where the image is square with side lengths that are a power of 2, that is 8 pixels or more.

The generator is parameterised based on the image size, with more upscale/transpose convolutions for more powers of 2.

My assumption is that larger images require more complex models in the discriminator. However, I have not yet found a way to effectively parameterise the discriminator to scale with image size. For now it still works OK for me up to 128, which is as high as I can do with my limited GPU memory.

Other small changes:

- Removes the hardcoded requirement for png images to be input. Maybe don't accept this PR if you don't want people's existing command line args to break (now just uses the glob, rather than accepting a directory).
 - Fixes directory creation on python 2.7
 - Deleted one or two lines of commented code.
 - Turned data shuffling back on.

I won't be offended if you don't accept this PR :) I'm not even close to being an expert. Thank you so much for the original code!

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -57,14 +61,12 @@ def __init__(self, sess, image_size=64, is_crop=False,
         self.c_dim = c_dim
 
         # batch normalization : deals with poor initialization helps gradient flow
-        self.d_bn1 = batch_norm(name='d_bn1')
-        self.d_bn2 = batch_norm(name='d_bn2')
-        self.d_bn3 = batch_norm(name='d_bn3')
+        self.d_bns = [
+            batch_norm(name='d_bn%d' % (i,)) for i in range(4)]
```
Comment (bamos) [inline model.py]: Let's use `.format` instead

File: model.py [hunk 2]
```diff
@@ -131,9 +133,10 @@ def build_model(self):
         self.complete_loss = self.contextual_loss + self.lam*self.perceptual_loss
         self.grad_complete_loss = tf.gradients(self.complete_loss, self.z)
 
+
     def train(self, config):
-        data = glob(os.path.join(config.dataset, "*.png"))
-        #np.random.shuffle(data)
+        data = glob(config.dataset)
```
Comment (bamos) [inline model.py]: Can you modify this (and the one below) to keep the old behavior while also adding the new behavior you want? If you want to support other image extensions, add them to the glob

**Comment by bamos** [comment]
Thanks! This mostly looks great. I added a few minor review comments we should change before merging this in

**Comment by md5wasp** [comment]
I think maybe I need to l2github, because I can't quite find your comments!

**Review by bamos — CHANGES_REQUESTED** [review]

**Comment by bamos** [comment]
Ah oops, I didn't realize that I needed to submit the comments for you to see them. :)

File: model.py (hunk 2 above)
Comment (md5wasp) [reply model.py]: OK, I've put it back to just accepting a folder and inserting the glob in the code, but now it supports png and jpg/jpeg. Currently that's hardcoded, would you prefer the supported file types to be a parameter to the constructor instead?

**Comment by bamos** [comment]
Thanks! This looks great now.
